app/src/main/java/com/github/se/eventradar/viewmodel/EventsOverviewViewModel.kt

```diff
@@ -288,6 +288,11 @@ constructor(
     // Reset search and filter when tab is changed
     state.value =
         state.value.copy(
+            eventList =
```

Comment:
could u please explain wjyu within the state.value.copy, you have that...

upcomingEventList and upcomingEventList  = to a state.value.copy...?

I tried going on ur branch to make it more optimal so i wanted to aks if there is some reason for this that I am not getting

Reply:
Good point, I have updated to use the currentState.update value instead of copy.